garrysmod/lua/includes/modules/outline.lua

```diff
@@ -0,0 +1,117 @@
+
+module("outline",package.seeall)
+
+local List = {}
+local RenderEnt = NULL
+
+local CopyMat		= Material("pp/copy")
+local OutlineMat	= CreateMaterial("OutlineMat","UnlitGeneric",{["$ignorez"] = 1,["$alphatest"] = 1})
+local StoreTexture	= render.GetScreenEffectTexture(0)
+local DrawTexture	= render.GetScreenEffectTexture(1)
+
+--[[-------------------------------
+	Modes:
+	0 - Always draw
+	1 - Draw if not visible
+	2 - Draw if visible
+--]]-------------------------------
+
+function Add(ents,color,mode)
+	if !istable(ents) then ents = {ents} end	--Support for passing Entity as first argument
+	if table.IsEmpty(ents) then return end		--Do not pass empty tables
+	if #List>=255 then return end				--Maximum 255 reference values
+	
+	table.insert(List,{
```

Comment:
You can use numerical/enum keys instead of string keys here for more efficient access. So you would add
```
local ENTS = 1
local COLOR = 2
local MODE = 3
```
and use those for indexing.

Reply:
is it so important? string keys used in halo module

Reply:
Just a small optimisation since halos are already so taxing.